js/private/js_run_devserver.mjs

```diff
@@ -54,42 +53,128 @@ function isNodeModulePath(srcPath) {
     return false
 }
 
+// Determines if a file path is a 1p dep in the virtual store.
+// See js/private/test/js_run_devserver/js_run_devserver.spec.mjs for examples.
+export function is1pVirtualStoreDep(p) {
```

Comment:
Can this full method just be replaced with a regex?

Reply:
Hmm. With a forward reference to a matching group potentially. I'll try before landing.

Reply:
Only if it ends up simpler then the method 🤷

Reply:
Not sure if its clearer but definitely is shorter:
```
export function is1pVirtualStoreDep(p) {
    const unscoped1p =
        /^.+\/\.aspect_rules_js\/([^@\/]+)@0\.0\.0\/node_modules\/\1$/
    const scoped1p =
        /^.+\/\.aspect_rules_js\/@([^@+\/]+)\+([^@+\/]+)@0\.0\.0\/node_modules\/@\1\/\2$/
    return p.match(unscoped1p) || p.match(scoped1p)
}
```